### PR TITLE
ToCode Solution: תרגול: מבוא לשורת הפקודה

### PR DESCRIPTION
**Pull request by dovron**

### CODE REVIEW

File: 07_cmd_lab/cmdlab.txt [hunk 1]
```diff
@@ -3,11 +3,26 @@ Questions online: https://www.tocode.co.il/bundles/unix1/lessons/cmdlab
...
+  
+	echo @SHELL
+  
   - a list of all shells currently installed on the machine?
+  
+	cat /etc/shells
 
 2. View the file /etc/passwd on your machine with the command "cat /etc/passwd". Then read about the file in "man 5 passwd". What information is held in the file? What are the fields in each line?
 
+	the information is all the user acounts + password in the OS
```
Comment (vitalipe) [inline 07_cmd_lab/cmdlab.txt:19]: correct! :)

just as a side note: modern linux/unix systems storing password hashed in ```/etc/shadow```.. 

so passwords here are ignored (it's for backward compatibility)

File: 07_cmd_lab/cmdlab.txt [hunk 2]
```diff
@@ -3,11 +3,26 @@ Questions online: https://www.tocode.co.il/bundles/unix1/lessons/cmdlab
...
+	the fields are (seperated by ":") login name, encrypted password, user ID, group ID, user name, home directory and user interpreter 
+
```
Comment (vitalipe) [inline 07_cmd_lab/cmdlab.txt:21]: "user shell" if you want to be extra precise.

also it's cool to note that some users have: ```/bin/false``` or ```/bin/nologin``` it means that they can't login.. 

the nologin option will also print a message, try typing ```nologin``` from the command line to see it :smiley:

File: 07_cmd_lab/cmdlab.txt [hunk 3]
```diff
@@ -3,11 +3,26 @@ Questions online: https://www.tocode.co.il/bundles/unix1/lessons/cmdlab
...
 3. In Unix, the /etc folder holds configuration files. Display all files in "/etc" folder ordered by file size (Hint: search in "man ls")
+	
+	ll -S
```
Comment (vitalipe) [inline 07_cmd_lab/cmdlab.txt:24]: sweet :) btw bonus question what is ```ll``` and where it's defined?
hint: it's not a binary :)

File: 07_cmd_lab/cmdlab.txt [hunk 4]
```diff
@@ -3,11 +3,26 @@ Questions online: https://www.tocode.co.il/bundles/unix1/lessons/cmdlab
...
 
 4. Print current date/time formatted as: "Sun 21/03/2010, 14:30". Hint: read "man date" and search for the correct format string.
+
+	date +%a' '%x,' '%R
```
Comment (vitalipe) [inline 07_cmd_lab/cmdlab.txt:28]: it's somewhat a style pref, but consider using a single double queted string: ```date "+%a %x, %R"``` 

it's easier to   have multiple spaces ```"         x    y       z "``` and it's also a good practice for shell scripting...

**Comment by vitalipe** [comment]
good job! :)

**Comment by vitalipe** [comment]
wait? can't I approve without merging? @ynonp help plz :P

File: 07_cmd_lab/cmdlab.txt (hunk 3 above)
Comment (dovron) [reply 07_cmd_lab/cmdlab.txt:24]: alias ll='ls -alF'
:1st_place_medal:

Comment (vitalipe) [reply 07_cmd_lab/cmdlab.txt:24]: yeah! GJ! :smile: 
(you can use the builtin command ```type```)

**Pull request closed without merging**